File: web/src/lib/admin/KycSubmission.ts
export type KycSubmission = {
  id: number;
  user_id: number;
  status: string;
  legal_name: string;
  country_code: string;
  document_type: string;
  document_last4: string | null;
  evidence_reference: string | null;
  provider_reference: string | null;
  reviewer_user_id: number | null;
  rejection_reason: string | null;
  submitted_at: string;
  reviewed_at: string | null;
  created_at: string;
  updated_at: string;
};

File: src/infra/postgres/models/kyc_submission.rs
use crate::infra::postgres::schema::kyc_submissions;
use chrono::{DateTime, Utc};
use diesel::prelude::*;
use serde::Serialize;

#[derive(Queryable, Identifiable, Selectable, Debug, Clone, Serialize)]
#[diesel(table_name = kyc_submissions)]
pub struct KycSubmission {
    pub id: i64,
    pub user_id: i32,
    pub status: String,
    pub legal_name: String,
    pub country_code: String,
    pub document_type: String,
    pub document_last4: Option<String>,
    pub evidence_reference: Option<String>,
    pub provider_reference: Option<String>,
    pub reviewer_user_id: Option<i32>,
    pub rejection_reason: Option<String>,
    pub submitted_at: DateTime<Utc>,
    pub reviewed_at: Option<DateTime<Utc>>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Insertable, Debug)]
#[diesel(table_name = kyc_submissions)]
pub struct NewKycSubmission {
    pub user_id: i32,
    pub status: String,
    pub legal_name: String,
    pub country_code: String,
    pub document_type: String,
    pub document_last4: Option<String>,
    pub evidence_reference: Option<String>,
    pub provider_reference: Option<String>,
}
